### ISSUE TITLE
coverage reports a blank line as "missing" under pypy3.6 (pypy-7.1.1-beta)

### ISSUE THREAD

**Issue by mgedmin** [coveragepy/coveragepy#839]
**Describe the bug**

Here's a Travis CI job that measures my test coverage on pypy3 and reports line 130 is uncovered:
https://travis-ci.org/mgedmin/ghcloneall/jobs/571894288

Here's line 130 of the relevant source file:
https://github.com/mgedmin/ghcloneall/blob/d59785b5a46e342648aafb1adc87e8d1f75171bf/ghcloneall.py#L130

It's a blank line between two method definitions.

**To Reproduce**
1. Get pypy3.6 version 1.7.1-beta from https://pypy.org/download.html
2. Make sure `pypy3` will run this version of PyPy
3. git clone https://github.com/mgedmin/ghcloneall
4. cd ghcloneall && tox -e coverage-pypy3

Package versions: coverage-4.5.4 coverage-python-version-0.1.0

**Expected behavior**

100% coverage like it shows on CPython 2.7 through 3.7.

**Additional context**

I cannot actually reproduce this bug on my local laptop for the silliest reason imaginable: I'm on Ubuntu 19.04 and pypy3.6 doesn't run there (missing libssl.so.1.0.0 -- which is an old version no longer shipped by Ubuntu).

**Comment by nedbat** [comment]
Hmm, running on master gives different but also baffling results:
```
$ git log -1 --oneline
 1e67a7c (HEAD -> master, origin/master, origin/HEAD) Back to development: 1.9.2
$ tox -e coverage-pypy3
GLOB sdist-make: /private/tmp/ghcloneall/setup.py
coverage-pypy3 inst-nodeps: /private/tmp/ghcloneall/.tox/.tmp/package/1/ghcloneall-1.9.2.dev0.zip
coverage-pypy3 installed: You are using pip version 10.0.1, however version 19.2.3 is available.,You should consider upgrading via the 'pip install --upgrade pip' command.,atomicwrites==1.3.0,attrs==19.2.0,certifi==2019.9.11,cffi==1.12.3,chardet==3.0.4,coverage==4.5.4,coverage-python-version==0.1.0,ghcloneall==1.9.2.dev0,greenlet==0.4.13,idna==2.8,importlib-metadata==0.23,more-itertools==7.2.0,packaging==19.2,pluggy==0.13.0,py==1.8.0,pyparsing==2.4.2,pytest==5.2.1,readline==6.2.4.1,requests==2.22.0,requests-cache==0.5.2,six==1.12.0,urllib3==1.25.6,wcwidth==0.1.7,zipp==0.6.0
coverage-pypy3 run-test-pre: PYTHONHASHSEED='4062859472'
coverage-pypy3 run-test: commands[0] | coverage run -m pytest
================================== test session starts ===================================
platform darwin -- Python 3.6.1[pypy-7.1.1-beta], pytest-5.2.1, py-1.8.0, pluggy-0.13.0
cachedir: .tox/coverage-pypy3/.pytest_cache
rootdir: /private/tmp/ghcloneall, inifile: pytest.ini, testpaths: tests.py
collected 64 items

tests.py ................................................................          [100%]

=================================== 64 passed in 4.94s ===================================
coverage-pypy3 run-test: commands[1] | coverage report -m --fail-under=100
Name            Stmts   Miss  Cover   Missing
---------------------------------------------
ghcloneall.py     551      3    99%   190, 229, 235
ERROR: InvocationError for command /private/tmp/ghcloneall/.tox/coverage-pypy3/bin/coverage report -m --fail-under=100 (exited with code 2)
________________________________________ summary _________________________________________
ERROR:   coverage-pypy3: commands failed
```
Line 190 is a blank line, and 235 is in the middle of a comment!

Also, going back to the commit you were at when you wrote the bug gives the results you reported:
```
$ git checkout 56f9456
HEAD is now at 56f9456 Coverage measurement under pypy3 is bugged
$ git log -1 --oneline --format=oneline
 56f9456 (HEAD) Coverage measurement under pypy3 is bugged

$ tox -e coverage-pypy3
GLOB sdist-make: /private/tmp/ghcloneall/setup.py
coverage-pypy3 inst-nodeps: /private/tmp/ghcloneall/.tox/.tmp/package/1/ghcloneall-1.7.1.dev0.zip
coverage-pypy3 installed: You are using pip version 10.0.1, however version 19.2.3 is available.,You should consider upgrading via the 'pip install --upgrade pip' command.,atomicwrites==1.3.0,attrs==19.2.0,certifi==2019.9.11,cffi==1.12.3,chardet==3.0.4,coverage==4.5.4,coverage-python-version==0.1.0,ghcloneall==1.7.1.dev0,greenlet==0.4.13,idna==2.8,importlib-metadata==0.23,more-itertools==7.2.0,packaging==19.2,pluggy==0.13.0,py==1.8.0,pyparsing==2.4.2,pytest==5.2.1,readline==6.2.4.1,requests==2.22.0,requests-cache==0.5.2,six==1.12.0,urllib3==1.25.6,wcwidth==0.1.7,zipp==0.6.0
coverage-pypy3 run-test-pre: PYTHONHASHSEED='1378011209'
coverage-pypy3 run-test: commands[0] | coverage run -m pytest
================================== test session starts ===================================
platform darwin -- Python 3.6.1[pypy-7.1.1-beta], pytest-5.2.1, py-1.8.0, pluggy-0.13.0
cachedir: .tox/coverage-pypy3/.pytest_cache
rootdir: /private/tmp/ghcloneall, inifile: pytest.ini, testpaths: tests.py
collected 56 items

tests.py ........................................................                  [100%]

=================================== 56 passed in 4.35s ===================================
coverage-pypy3 run-test: commands[1] | coverage report -m --fail-under=100
Name            Stmts   Miss  Cover   Missing
---------------------------------------------
ghcloneall.py     465      1    99%   130
ERROR: InvocationError for command /private/tmp/ghcloneall/.tox/coverage-pypy3/bin/coverage report -m --fail-under=100 (exited with code 2)
________________________________________ summary _________________________________________
ERROR:   coverage-pypy3: commands failed
```

**Comment by mgedmin** [comment]
I spent way more time than reasonable wrestling LXD containers so I could get a working pypy3.6 on my laptop.  I then tried to compare

   pypy3.6 -m dis ghcloneall.py

with

  python3.6 -m dis ghcloneall.py

and if I'm not mistaken (and the 1st column displays line numbers), then pypy3.6 has some weird ideas about what line numbers correspond to what code.  More precisely, the line numbers differ from the correct ones by 256 exactly, then a few by 512, then the last ones by 768.

I'm going to go and file a pypy bug.
